jobs/performance_tests/templates/run_perf_test.erb

```diff
@@ -3,47 +3,70 @@
 set -e
 
 LOG_DIR=/var/vcap/sys/log/performance_tests
+GOSTATIC_BACKEND_HOST=<%= link("static").instances[0].address %>
+GOROUTER_PROXY_URI=<%= p("performance_tests.protocol") %>://${1}:<%= p("performance_tests.port") %>
+GOROUTER_HTTPS_PROXY_URI=https://${1}
 
-RPS=0
-LATENCY_90=0
-LATENCY_95=0
-LATENCY_99=0
-NUM_OK="could not extract status codes of responses from summary"
-
-BACKEND_ADDRESS=<%= link("static").instances[0].address %>
 RPS_LOWER_LIMIT=<%= p("performance_tests.throughput_lower_limit") %>
 LATENCY_UPPER_LIMIT_90=<%= p("performance_tests.latency_upper_limit_90") %>
 LATENCY_UPPER_LIMIT_95=<%= p("performance_tests.latency_upper_limit_95") %>
 LATENCY_UPPER_LIMIT_99=<%= p("performance_tests.latency_upper_limit_99") %>
+
 TOTAL_REQUESTS=<%= p("performance_tests.num_requests") %>
 TOTAL_CONCURRENT=<%= p("performance_tests.concurrent_requests") %>
-URI=<%= p("performance_tests.protocol") %>://${1}:<%= p("performance_tests.port") %>
-ROUTER=${2}
+
+
+RPS=0
+LATENCY_90=0
+LATENCY_95=0
+LATENCY_99=0
+NUM_OK="could not extract status codes of responses from summary"
 
 ensure_log_dir() {
   mkdir -p ${LOG_DIR}
 }
 
 run_perf_tests() {
   date=$(date +%s)
-  num_loops=$((<%= p("performance_tests.num_requests") %>/<%=p("performance_tests.concurrent_requests")%>))
 
-  # run load test
-  echo "Starting routed load test of ${TOTAL_REQUESTS} requests, ${TOTAL_CONCURRENT} concurrent..."
+  # run load test against gorouter using http1
+  echo "Starting gorouter http1 load test of ${TOTAL_REQUESTS} requests, ${TOTAL_CONCURRENT} concurrent..."
   exec /var/vcap/packages/hey/bin/hey \
     -n $TOTAL_REQUESTS \
     -c $TOTAL_CONCURRENT \
     -t 0 \
     -host "<%=p("performance_tests.host") %>" \
-    ${URI} | tee ${LOG_DIR}/routed_loadtest_${date}.log
+    ${GOROUTER_PROXY_URI} | tee ${LOG_DIR}/routed_loadtest_${date}.log
   echo ""
 
-  echo "Starting direct load test of ${TOTAL_REQUESTS} requests, ${TOTAL_CONCURRENT} concurrent..."
+  # run load test against gorouter using http2
+  echo "Starting gorouter http2 load test of ${TOTAL_REQUESTS} requests, ${TOTAL_CONCURRENT} concurrent..."
   exec /var/vcap/packages/hey/bin/hey \
     -n $TOTAL_REQUESTS \
     -c $TOTAL_CONCURRENT \
     -t 0 \
-    http://${BACKEND_ADDRESS}:8080 | tee ${LOG_DIR}/direct_loadtest_${date}.log
+    -h2 \
+    -host "<%=p("performance_tests.host") %>" \
+    ${GOROUTER_HTTPS_PROXY_URI} | tee ${LOG_DIR}/routed_loadtest_${date}.log
+  echo ""
+
+  # run load test aginst gostatic envoy
+  echo "Starting gostatic envoy test of ${TOTAL_REQUESTS} requests, ${TOTAL_CONCURRENT} concurrent..."
```

Comment:
❓ Can you explain what this is testing? What is a "gostatic envoy" (this line) vs a "gostatic direct load test" (below)

Reply:
"gostatic envoy" hits the app through envoy using HTTP/1.1 and TLS. "gostatic direct load test" hits the app directly via HTTP/1.1.

The purpose IIRC was to determine how much envoy TLS termination was slowing down our requests.

Reply:
Oh, nice. That makes a lot of sense. Can you add something to that effect in the comments so it is more clear?